resourcemanager/commonids/community_gallery_image.go

```diff
@@ -0,0 +1,112 @@
+package commonids
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/resourceids"
+)
+
+var _ resourceids.ResourceId = &CommunityGalleryImageId{}
+
+// CommunityGalleryImageId is a struct representing the Resource ID for a Community Gallery Image
+type CommunityGalleryImageId struct {
+	CommunityGalleryName string
+	ImageName            string
+}
+
+// NewCommunityGalleryImageID returns a new CommunityGalleryImageId struct
+func NewCommunityGalleryImageID(communityGallery string, image string) CommunityGalleryImageId {
+	return CommunityGalleryImageId{
+		CommunityGalleryName: communityGallery,
+		ImageName:            image,
+	}
+}
+
+// ParseCommunityGalleryImageID parses 'input' into a CommunityGalleryImageId
+func ParseCommunityGalleryImageID(input string) (*CommunityGalleryImageId, error) {
+	parser := resourceids.NewParserFromResourceIdType(&CommunityGalleryImageId{})
+	parsed, err := parser.Parse(input, false)
+	if err != nil {
+		return nil, fmt.Errorf("parsing %q: %+v", input, err)
+	}
+
+	id := CommunityGalleryImageId{}
+	if err = id.FromParseResult(*parsed); err != nil {
+		return nil, err
+	}
+
+	return &id, nil
+}
+
+// ParseCommunityGalleryImageIDInsensitively parses 'input' case-insensitively into a CommunityGalleryImageId
+// note: this method should only be used for API response data and not user input
+func ParseCommunityGalleryImageIDInsensitively(input string) (*CommunityGalleryImageId, error) {
+	parser := resourceids.NewParserFromResourceIdType(&CommunityGalleryImageId{})
+	parsed, err := parser.Parse(input, true)
+	if err != nil {
+		return nil, fmt.Errorf("parsing %q: %+v", input, err)
+	}
+
+	id := CommunityGalleryImageId{}
+	if err = id.FromParseResult(*parsed); err != nil {
+		return nil, err
+	}
+
+	return &id, nil
+}
+
+func (id *CommunityGalleryImageId) FromParseResult(input resourceids.ParseResult) error {
+
+	var ok bool
+
+	if id.CommunityGalleryName, ok = input.Parsed["communityGalleryName"]; !ok {
+		return resourceids.NewSegmentNotSpecifiedError(id, "communityGalleryName", input)
+	}
+
+	if id.ImageName, ok = input.Parsed["imageName"]; !ok {
+		return resourceids.NewSegmentNotSpecifiedError(id, "imageName", input)
+	}
+
+	return nil
+}
+
+// ValidateCommunityGalleryImageID checks that 'input' can be parsed as a Community Gallery Image ID
+func ValidateCommunityGalleryImageID(input interface{}, key string) (warnings []string, errors []error) {
+	v, ok := input.(string)
+	if !ok {
+		errors = append(errors, fmt.Errorf("expected %q to be a string", key))
+		return
+	}
+
+	if _, err := ParseCommunityGalleryImageID(v); err != nil {
+		errors = append(errors, err)
+	}
+
+	return
+}
+
+// ID returns the formatted Community Gallery Image ID
+func (id CommunityGalleryImageId) ID() string {
+	fmtString := "/communityGalleries/%s/images/%s"
```

Comment:
Since we're implementing the other ID Types, do we also need one for `Community Gallery ID` too?

Reply:
I'll open a new PR for this 👍